cogs/strawpoll.py

```diff
@@ -0,0 +1,292 @@
+        self.poll_length = poll_length
+        self.start_time = time()
+
+    def _bar_creator(self, data, option):
+        display_bar = " :: "
+        bar_length = self.settings['bar_length']
+        for y in range(bar_length):
+            if (max(data['votes']) > 0):
+                pct = float(
+                    data['votes'][option])/float(max(data['votes']))
+            else:
+                continue
+            if (pct != 0 and pct >= float(y)/float(bar_length-1)):
+                display_bar += '|'
+        return display_bar
+
+    async def update_results(self):
+        data = _get_poll(self.poll_id)
+        if data is None:
+            await self.bot.edit_message(self.message, 
+                embed=discord.Embed(title="Error receiving strawpoll data!"))
+        embed=discord.Embed(title=unescape(data['title']), 
+            url='https://strawpoll.me/' + str(self.poll_id), 
+            description="Strawpoll Results", color=self.author.color)
+        embed.set_author(
+            name=self.author.nick if self.author.nick else self.author.name, 
+            icon_url=self.author.avatar_url)
+        for option in range(len(data['options'])):
+            embed.add_field(inline=False, 
+                name="{} ({} Vote{})".format(
+                    unescape(data['options'][option]),
+                    str(data['votes'][option]),
+                    '' if (data['votes'][option] == 1) else 's'),
+                value=self._bar_creator(data, option))
+        time_left = round(self.poll_length-(time()-self.start_time))
+        if (time_left > 0):
+            embed.set_footer(text="{} sec{} left".format(
+                str(time_left), '' if (time_left == 1) else 's'))
+        else: 
+            embed.set_footer(text="as of {}".format(
+                strftime('%A %B %-m, %Y @ %I:%M:%S%p ')))
+        await self.bot.edit_message(self.message, embed=embed)
+
+class Strawpoll:
+    """Create, link, and update live results for a Strawpoll within Discord"""
+    
+    def __init__(self, bot, settings_path):
+        self.bot = bot
+        self.settings_path = settings_path
+        self.poll_sessions = []
+        self.poll_session_tasks = {}
+        try:
+            if not dataIO.is_valid_json(settings_path):
+                raise FileNotFoundError("JSON file not valid")
+            settings = dataIO.load_json(settings_path)
+            self.settings = defaultdict(dict, settings)
+        except FileNotFoundError:
+            self.settings = {}
+            if not os.path.exists(os.path.dirname(settings_path)):
+                print("Creating strawpoll data folder...")
+                os.makedirs(os.path.dirname(settings_path))
+            dataIO.save_json(settings_path, self.settings)
+
+    def _new_server_settings(self, server_id):
+        default_settings = {
+            'refresh_emoji': '🔄',
+            'poll_length': 60.0,
+            'poll_react_time': 1800.0, # 30 minutes
```

Comment:
Seems a little long but I suppose it's ok. I'd go for something a little more conservative, like 5 or 10 minutes.

Reply:
5 minutes is probably a good default; a lot can happen in 30 minutes in a social context